test/main.cc

```diff
@@ -7,5 +7,14 @@ int main(int argc, char** argv) {
   ::setenv("TEST_RUNDIR", (TestEnvironment::getCheckedEnvVar("TEST_SRCDIR") + "/" +
                            TestEnvironment::getCheckedEnvVar("TEST_WORKSPACE")).c_str(),
            1);
+  // Select whether to test only for IPv4, IPv6, or both. The default is to
+  // test for both. Options are {"v4only", "v6only", "all"}. Set IP_TEST_TYPE
+  // to "v4only" if the system currently does not support IPv6 network
+  // operations. Similary set IP_TEST_TYPE to "v6only" if IPv4 has already been
+  // phased out of network operations. Set to "all" (or don't set) if testing both
+  // v4 and v6 addresses is desired. This feature will be rolled out to all tests
+  // in upcoming PRs but currently only applies to listen_socket_impl_test
+  // and listener_impl_test tests.
+  ::setenv("IP_TEST_TYPE", "all", 1);
```

Comment:
Replace 1 with 0 here, since this will avoid overwriting if someone has set this externally as a forced override, which is behavior we want to support (e.g. someone running the tests on an IPv4 only system can set `--test_env=ENVOY_IP_TEST_VERSIONS=v4only` on the Bazel command-line.